utils/src/linked_list.c

```diff
@@ -0,0 +1,230 @@
+#include "linked_list.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+/**
+ * @brief      Prints data of a linked list.
+ *
+ * @param[in]      list  The list to print
+ */
+void print_linked_list(linked_list_t *list) {
+    // Check for NULL
+    if (list == NULL) {
+        return;
+    }
+    node_t *curr_node = NULL;
+    for (curr_node = list->head; curr_node != NULL;
+         curr_node = curr_node->next) {
+        printf("%s\n", (char *)curr_node->data);
+    }
+}
+
+/**
+ * @brief      Initializes the linked list
+ *
+ * @return     Returns an empty linked list
+ */
+linked_list_t *init_list() {
+    linked_list_t *list = (linked_list_t *)calloc(1, sizeof(linked_list_t));
+
+    // Check NULL
+    if (list == NULL) {
+        return NULL;
+    }
+    return list;
+}
+
+/**
+ * @brief      Returns the data from the tail of the list
+ *
+ * @param[in]      list  The list
+ *
+ * @return     Returns the last item in the list
+ */
+void *get_tail(linked_list_t *list) {
```

Comment:
Is this just for access control?  If so the name of ``tail`` field of ``linked_list_t`` should probably be renamed to ``_tail``.  Also, using this function doesn't seem to simplfy the code much, as it still is going to require checks for ``NULL`` on the value it returns.

Reply:
@garfr Will change the `tail` to `_tail`. The purpose is to abstract the underlying access from other structures that will be using this list (like the queue for example that uses this for the `front` function